web/src/bootstrap_typeahead.ts

```diff
@@ -731,6 +731,9 @@ export class Typeahead<ItemType extends string | object> {
         switch (pseudo_keycode) {
             case 40: // down arrow
             case 38: // up arrow
+            case 16:
+            case 17:
+                // ctrl and shift keys should not affect the typeahead
                 break;
```

Comment:
Let's make a separate commit for this. Also, we need to include key for `command` here since that's what is used in mac. Also the below should take into account command + shift instead of ctrl + shift if user is using a mac.

Reply:
This would conflict with the note written above
```
        // NOTE: Ideally we can ignore meta keyup calls here but
        // it's better to just trigger the lookup call to update the list in case
        // it did modify the query. For example, `Command + delete` on Mac
        // doesn't trigger a keyup event but when `Command` is released, it
        // triggers a keyup event which correctly updates the list.
```

since to prevent the typeahead from disappearing, we'd need to ignore the meta keys

Reply:
One solution would be to keep track of the last key combination pressed, and when encountering the meta key, use that info to decide whether to ignore it or not. 
So we ignore the meta key if we had just pressed `Cmd+Shift+C` and we don't if we just pressed `Cmd +Delete`.

Reply:
I see, looks like we need to fix how we handle keyup / keydown / keypress events. It's not super clearly thoughtout right now. I will open a PR for this. Then we can proceed further on this.